packages/belos/src/BelosTypes.cpp

```diff
@@ -189,5 +189,14 @@ namespace Belos {
     }
   }
 
+  // Initialize DefaultSolverParameters.  Has to be done this way because
```

Comment:
Could you be more specific about these linking errors?  I haven't seen those with other static constexpr stuff in Belos.

Reply:
See [here](https://stackoverflow.com/questions/26196095/static-constexpr-odr-used-or-not).  In short, if a static constexpr is used as an argument to a function, it may be odr-used (depending on the compiler and optimization), in which case the constant must have a definition (and `static constexpr double blah = ...;` is not a definition).

Reply:
Btw, this is a problem with C++11 that was fixed in C++17.

Reply:
@etphipp I'm glad C++17 fixed that :-D